src/CreateEntityMappingView/CreateEntityMappingView.js

```diff
@@ -249,15 +304,140 @@ class CreateEntityMappingView extends ConnectionComponent {
 			</div>);
 	}
 
+	// is called to load the existing entity mapping in the edit view
+	entityMappingPending() {
+		if (!this.props.entityMapping) {
+			return <LoadingAnimation/>;
+		}
+		const entityMappingConnection = super.render(this.props.entityMapping);
+		if (entityMappingConnection) {
+			return entityMappingConnection;
+		}
+		if (!this.props.entityMapping.value) {
+			return <LoadingAnimation/>;
+		}
+		return null;
+	}
+
+	componentWillMount() {
+		if (!this.isCreateView) {
+			this.props.lazyLoadEntityMapping();
+		}
+	}
+
 	render() {
+		if (!this.isCreateView) {
+			return this.renderEditView();
+		}
+		return this.renderCreateView();
+	}
+
+	renderEditView() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Function has a complexity of 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AFunctionComplexity)